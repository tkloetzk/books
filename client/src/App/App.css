section {
  margin-top: 115px;
}

#container {
  display: 'flex';
}
