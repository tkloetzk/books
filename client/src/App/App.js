import React from 'react';
import './App.css';
import Header from './header/Header';
import Search from './Search/Search';
import AppBar from '@material-ui/core/AppBar';
import Bookshelf from './Bookshelf/Bookshelf';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import SwipeableViews from 'react-swipeable-views';
import Notification from './Notification/Notification';
import { connect } from 'react-redux';
import { LOADING_STATUSES } from '../util/constants';
import { withStyles } from '@material-ui/core';

const styles = {
  header: {
    backgroundColor: 'white',
    height: 60,
    backgroundSize: 'cover',
    left: 0,
    right: 0,
    position: 'fixed',
    top: 0,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    borderBottom: '2px solid',
  },
};
class App extends React.Component {
  state = {
    value: 0,
    open: true,
  };

  componentDidUpdate(prevProps) {
    const { saveStatus } = this.props;
    if (saveStatus === LOADING_STATUSES.success) {
      this.setState({ open: true });
    }
  }

  handleChange = (event, value) => {
    this.setState({ value });
  };

  handleChangeIndex = index => {
    this.setState({ value: index });
  };

  handleClose = (event, reason) => {
    if (reason === 'clickaway') {
      return;
    }

    this.setState({ open: false });
  };
  render() {
    const { value, open } = this.state;
    const { saveStatus, classes } = this.props;
    return (
      <div className="App">
        <Header>Book Review Aggregator</Header>
        <section className="container">
          <AppBar
            position="static"
            color="default"
            style={{ marginTop: '-13px', marginBottom: '14px' }}
          >
            <Tabs
              value={value}
              onChange={this.handleChange}
              indicatorColor="primary"
              textColor="primary"
              variant="fullWidth"
            >
              <Tab label="Search" />
              <Tab label="Bookshelf" />
            </Tabs>
          </AppBar>
          <SwipeableViews
            // axis={theme.direction === 'rtl' ? 'x-reverse' : 'x'}
            index={value}
            onChangeIndex={this.handleChangeIndex}
          >
            <Search />
            <Bookshelf />
          </SwipeableViews>
          <Notification
            open={open}
            handleClose={this.handleClose}
            autoHideDuration={4000}
            message={saveStatus.message}
            type={saveStatus.status}
          />
        </section>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    saveStatus: state.bookshelf.saveStatus,
  };
};

export default connect(mapStateToProps)(withStyles(styles)(App));
