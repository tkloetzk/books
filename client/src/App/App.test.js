import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';

describe('App', () => {
  let props;
  let wrapper;
  let instance;
  beforeEach(() => {
    props = {
      classes: {},
    };
    wrapper = shallow(<App {...props} />);
    instance = wrapper.instance();
  });
  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });
  describe('Switching Tab functionality', () => {
    it('Initial Tab index is 0', () => {
      expect(instance.state.value).toEqual(0);
    });
    it('handleChangeIndex changes tabs', () => {
      instance.handleChangeIndex(1);
      expect(instance.state.value).toEqual(1);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
