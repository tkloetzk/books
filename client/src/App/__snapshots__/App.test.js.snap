// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Switching Tab functionality handleChange changes tabs 1`] = `
<div
  className="App"
>
  <Header
    scrollAction={250}
  >
    Book Review Aggregator
  </Header>
  <section
    className="container"
  >
    <WithStyles(AppBar)
      color="default"
      position="static"
      style={
        Object {
          "marginBottom": "14px",
          "marginTop": "-13px",
        }
      }
    >
      <WithStyles(Tabs)
        indicatorColor="primary"
        onChange={[Function]}
        textColor="primary"
        value={1}
        variant="fullWidth"
      >
        <WithStyles(Tab)
          label="Search"
        />
        <WithStyles(Tab)
          label="Bookshelf"
        />
      </WithStyles(Tabs)>
    </WithStyles(AppBar)>
    <ReactSwipableView
      animateHeight={false}
      animateTransitions={true}
      axis="x"
      disableLazyLoading={false}
      disabled={false}
      enableMouseEvents={false}
      hysteresis={0.6}
      ignoreNativeScroll={false}
      index={1}
      resistance={false}
      springConfig={
        Object {
          "delay": "0s",
          "duration": "0.35s",
          "easeFunction": "cubic-bezier(0.15, 0.3, 0.25, 1)",
        }
      }
      style={
        Object {
          "maxHeight": "78vh",
        }
      }
      threshold={5}
    >
      <Connect(Search) />
      <Connect(Bookshelf)
        active={true}
      />
    </ReactSwipableView>
  </section>
</div>
`;

exports[`App renders 1`] = `
<div
  className="App"
>
  <Header
    scrollAction={250}
  >
    Book Review Aggregator
  </Header>
  <section
    className="container"
  >
    <WithStyles(AppBar)
      color="default"
      position="static"
      style={
        Object {
          "marginBottom": "14px",
          "marginTop": "-13px",
        }
      }
    >
      <WithStyles(Tabs)
        indicatorColor="primary"
        onChange={[Function]}
        textColor="primary"
        value={0}
        variant="fullWidth"
      >
        <WithStyles(Tab)
          label="Search"
        />
        <WithStyles(Tab)
          label="Bookshelf"
        />
      </WithStyles(Tabs)>
    </WithStyles(AppBar)>
    <ReactSwipableView
      animateHeight={false}
      animateTransitions={true}
      axis="x"
      disableLazyLoading={false}
      disabled={false}
      enableMouseEvents={false}
      hysteresis={0.6}
      ignoreNativeScroll={false}
      index={0}
      resistance={false}
      springConfig={
        Object {
          "delay": "0s",
          "duration": "0.35s",
          "easeFunction": "cubic-bezier(0.15, 0.3, 0.25, 1)",
        }
      }
      style={
        Object {
          "maxHeight": "78vh",
        }
      }
      threshold={5}
    >
      <Connect(Search) />
      <Connect(Bookshelf)
        active={false}
      />
    </ReactSwipableView>
  </section>
</div>
`;
